src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -7975,12 +7978,7 @@ private BoundExpression BindIndexedPropertyAccess(SyntaxNode syntax, BoundExpres
                             property.OriginalDefinition is { ParameterCount: 1 } original &&
                             original.Parameters[0] is { Type: { SpecialType: SpecialType.System_Int32 }, RefKind: RefKind.None })
                         {
-<<<<<<< HEAD
-=======
-                            CheckImplicitThisCopyInReadOnlyMember(receiverOpt, lengthOrCountProperty.GetMethod, diagnostics);
                             ReportDiagnosticsIfObsolete(diagnostics, property, syntax, hasBaseReceiver: false);
-                            ReportDiagnosticsIfObsolete(diagnostics, lengthOrCountProperty, syntax, hasBaseReceiver: false);
```

Comment:
Moved this check to caller (line 7934 above) for binding indexer.
For binding list-patterns we still have a PROTOTYPE marker tracking this in test `ListPattern_ObsoleteMembers`.